common/network/uri_parser.c

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2018 IOTA Stiftung
+ * https://github.com/iotaledger/entangled
+ *
+ * Refer to the LICENSE file for licensing information
+ */
+
+#include <stdlib.h>
+#include <string.h>
+
+#include "common/network/uri_parser.h"
+
+bool uri_parse(char const *const uri, char *const scheme, char *const host,
```

Comment:
You're not enforcing scheme/host lengths anywhere (or that they've been preallocated to the correct size before)

That's potentially dangerous and could be used for exploitation. Add `scheme_len` / `host_len` parameters maybe?